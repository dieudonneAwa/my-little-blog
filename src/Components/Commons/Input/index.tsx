import styled from "styled-components";
import { DefaultTheme } from "../../../Theme";

const StyledInput = styled.input<{ theme: DefaultTheme }>`
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  display: block;
  font-size: ${({ theme }) => theme.font.size.medium};
  border-radius: ${(p) => p.theme.radii.small};
  border: 1px solid ${(p) => p.theme.colors.cultured};
  background-color: ${(p) => p.theme.colors.lightgrey};
  padding: 1.1rem 1rem;
  margin: 0.5rem;
  outline: none;
  width: calc(100% - 1rem);
  box-sizing: border-box;
  &:disabled {
    background-color: rgba(128, 128, 128, 0.12);
  }

  &::placeholder {
    color: ${({ theme }) => theme.colors.c600};
  }
`;

const Input = (props: any) => <StyledInput {...props} />;

export default Input;
