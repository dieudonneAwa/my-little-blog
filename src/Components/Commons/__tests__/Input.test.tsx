/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import Input from '../Input';

describe('Input', (): void => {
  let wrapper: ShallowWrapper<void>;
  beforeEach((): void => {
    wrapper = shallow(<Input />);
  });

  it('should render Input component', (): void => {
    expect(wrapper).toMatchSnapshot();
  });
});
