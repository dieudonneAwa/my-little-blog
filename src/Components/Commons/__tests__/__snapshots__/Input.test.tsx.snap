// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render Input component 1`] = `<styled.input />`;
